packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -256,6 +263,50 @@ const HoverCardContentImpl = React.forwardRef<
   const popperScope = usePopperScope(__scopeHoverCard);
   const ref = React.useRef<HoverCardContentImplElement>(null);
   const composedRefs = useComposedRefs(forwardedRef, ref);
+  const [containTextSelection, setContainTextSelection] = React.useState(false);
+
+  React.useEffect(() => {
+    if (containTextSelection) {
+      const body = document.body;
+
+      // Safari requires prefix
+      originalBodyUserSelect = body.style.userSelect || body.style.webkitUserSelect;
+
+      body.style.userSelect = 'none';
+      body.style.webkitUserSelect = 'none';
+      return () => {
+        body.style.userSelect = originalBodyUserSelect;
+        body.style.webkitUserSelect = originalBodyUserSelect;
+      };
+    }
+  }, [containTextSelection]);
+
+  const { onDismiss } = context;
+
+  React.useEffect(() => {
+    if (ref.current) {
+      const handlePointerUp = (event: PointerEvent) => {
+        // Delay a frame to ensure we always access the latest selection
+        setTimeout(() => {
```

Comment:
I did a lot of testing between browsers and found that the selection would sometimes lag `pointerup` by a frame, I found delaying here to be the simplest of the options available as we're not handling any consumer events.